djautotask/api.py

```diff
@@ -76,23 +76,26 @@ def get_cached_url(cache_key):
     return cache.get(cache_key)
 
 
-def get_api_connection_url():
-    return _get_connection_url('url')
+def update_cache(json_obj):
+    cache.set('url', json_obj['url'])
+    cache.set('webUrl', json_obj['webUrl'])
 
 
-def get_web_connection_url():
-    return _get_connection_url('webUrl')
+def get_api_connection_url(force_fetch=False):
+    return _get_connection_url('url', force_fetch)
 
 
-def _get_connection_url(field):
-    cache_key = 'zone_info_' + field
-    api_url_from_cache = get_cached_url(cache_key)
+def get_web_connection_url(force_fetch=False):
+    return _get_connection_url('webUrl', force_fetch)
 
-    if not api_url_from_cache:
+
+def _get_connection_url(field, force_fetch=False):
+    api_url_from_cache = get_cached_url(field)
+
+    if not api_url_from_cache or force_fetch:
         try:
             json_obj = get_zone_info(settings.AUTOTASK_CREDENTIALS['username'])
             url = json_obj[field]
-            cache.set(cache_key, url, timeout=None)
```

Comment:
This is actually the timeout mentioned in the issue description "Done when" section. It shouldn't be removed, it should be updated to some finite amount of time. Let's say 10 minutes.